internal/venice-client-common/src/main/java/com/linkedin/venice/stats/VeniceMetricsConfig.java

```diff
@@ -0,0 +1,473 @@
+  /**
+   * Export opentelemetry metrics to a log exporter
+   * {@link VeniceOpenTelemetryMetricsRepository.LogBasedMetricExporter}
+   */
+  public static final String OTEL_VENICE_EXPORT_TO_LOG = "otel.venice.export.to.log";
+
+  /**
+   * Export opentelemetry metrics to {@link #OTEL_EXPORTER_OTLP_METRICS_ENDPOINT}
+   * over {@link #OTEL_EXPORTER_OTLP_METRICS_PROTOCOL}
+   */
+  public static final String OTEL_VENICE_EXPORT_TO_ENDPOINT = "otel.venice.export.to.endpoint";
+
+  /**
+   * Config Map to add custom dimensions to the metrics: Can be used for system dimensions
+   * amongst other custom dimensions <br>
+   * These will be emitted along with all the metrics emitted.
+   *
+   *
+   * custom dimensions are passed as key=value pairs separated by '='
+   * Multiple headers are separated by ','
+   * For example: "custom_dimension_one=value1,custom_dimension_two=value2,custom_dimension_three=value3"
+   */
+  public static final String OTEL_VENICE_CUSTOM_DIMENSIONS_MAP = "otel.venice.custom.dimensions.map";
+
+  /**
+   * Protocol over which the metrics are exported to {@link #OTEL_EXPORTER_OTLP_METRICS_ENDPOINT} <br>
+   * 1. {@link OtlpConfigUtil#PROTOCOL_HTTP_PROTOBUF}  => "http/protobuf" <br>
+   * 2. {@link OtlpConfigUtil#PROTOCOL_GRPC}  => "grpc"
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_PROTOCOL = "otel.exporter.otlp.metrics.protocol";
+
+  /**
+   * The Endpoint to which the metrics are exported
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_ENDPOINT = "otel.exporter.otlp.metrics.endpoint";
+
+  /**
+   * Additional headers to pass while creating OpenTelemetry exporter
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_HEADERS = "otel.exporter.otlp.metrics.headers";
+
+  /**
+   * Aggregation Temporality selector to export only the delta or cumulate or different
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE =
+      "otel.exporter.otlp.metrics.temporality.preference";
+
+  /**
+   * Default histogram aggregation to be used for all histograms: Select one of the below <br>
+   * 1. base2_exponential_bucket_histogram <br>
+   * 2. explicit_bucket_histogram
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION =
+      "otel.exporter.otlp.metrics.default.histogram.aggregation";
+
+  /**
+   * Max scale for base2_exponential_bucket_histogram
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION_MAX_SCALE =
+      "otel.exporter.otlp.metrics.default.histogram.aggregation.max.scale";
+
+  /**
+   * Max buckets for base2_exponential_bucket_histogram
+   */
+  public static final String OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION_MAX_BUCKETS =
+      "otel.exporter.otlp.metrics.default.histogram.aggregation.max.buckets";
+
+  private final String serviceName;
+  private final String metricPrefix;
+  private final Collection<MetricEntities> metricEntities;
```

Comment:
E.g. why can't this just be `Collection<MetricEntity>`?

Reply:
as discussed, removed `MetricEntities` interface completely and using `MetricEntity`